go/database/mpt/nodes.go

```diff
@@ -1505,6 +1531,9 @@ type AccountNode struct {
 	pathLength byte
 }
 
+func (n *AccountNode) HasEmptyStorage(NodeSource, []Nibble) (bool, error) {
+	return n.storage.Id().IsEmpty(), nil
```

Comment:
At this point you may have reached the wrong `AccountNode`. Without checking the address, you would not know.

For instance, if you create a MPT with a single account with address `A` and then add a single storage slot, and then ask whether account `B` (A!=B) has an empty storage, the answer probably is no, although `B` should have an empty storage.